account_wallet_sale/models/sale_order.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020 ACSONE SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    has_wallet = fields.Boolean(
+        compute="_compute_has_wallet",
+        help="This is used to check if sale order lines contain wallet",
+    )
+
+    def _reapply_wallet(self):
+        """
+        If sale order contains a line with wallet, save it, unlink the
+        corresponding lines and then apply wallet.
+        This is typically used when sale order lines have changed and we
+        need to check if the wallet is still applicable.
+        :return:
+        """
+        for sale in self.filtered(lambda s: s.state == "draft" and s.has_wallet):
+            wallet_lines = self.mapped("order_line").filtered("account_wallet_id")
```

Comment:
`wallet_lines = sale.mapped("order_line").filtered("account_wallet_id")`

Reply:
How to add acount wallet